### PR TITLE
Updated - Vanilla Zappa/Pyramid Project

### PR DESCRIPTION
**Pull request by Nico005**
Updated - Vanilla Zappa/Pyramid Project
Overview

Creates a PyZAPI project with python3.6 and a virtualenv.
Zappa will run within a configured virtualenv

Notes
To successfully run the project please refer to the Contributing.md file in the .github folder
A AWS MySQL connection is required aswell as a AWS IAM User account.

### CODE REVIEW

File: PyZAPI/zappa_settings.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "default": {
+        "app_function": "pyzapi.app.zappa_dev",
+        "aws_region": "us-east-1",
+        "profile_name": "default",
+        "s3_bucket": "zappa-0oqiolisl",
```
Comment (oakensoul) [inline PyZAPI/zappa_settings.json]: As I mentioned with the developer.ini etc., you don’t want to hard code the s3 bucket name in here, you probably want to create a zappa.json.example maybe? Then have folks add what they need in here?

I think these are good opportunities for using our script system… that being said, I’ll probably just let this pass through and then put a PR in myself with the changes to show you what I’m thinking.

File: ansible/roles/base/tasks/main.yml [hunk 2]
```diff
@@ -1,32 +1,27 @@
...
 
     - name: install python-pip
-      apt: name=python-pip
-      become: yes
+      apt: 
+       name: python3-pip
 
-    - name: install python-virtualenv
-      apt: name=python-virtualenv
-      become: yes
```
Comment (oakensoul) [inline ansible/roles/base/tasks/main.yml:29]: Does this actually work if you remove the become: yes statements?

Also, as I mentioned before, there are still a handful of spacing issues … and now weird capitalization changes, etc… lol. As with the other items, I’ll probably just let them through and fix them on my own just to get you onto the next items.

**Review by oakensoul — DISMISSED** [review]
I’m going to put an approval on this, just so we can start moving forward. There are still some items that are going to need to be tweaked, but it’s hard for me to tell how much they need to be tweaked without trying it out myself.

**Comment by oakensoul** [comment]
@markdia @ajbosco Please review this as well. I’m giving it a good enough for now, but please look to see if there’s anything I’m missing.

File: PyZAPI/pyzapi/__init__.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from pyramid.config import Configurator
+
+
+def main(global_config, **settings):
+    """ This function returns a Pyramid WSGI application.
+    """
+    config = Configurator(settings=settings)
+    config.include('pyramid_jinja2')
```
Comment (markdia) [inline PyZAPI/pyzapi/__init__.py]: jinja a choice as a templating lib or a default?

File: PyZAPI/pyzapi/views/default.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    try:
+        query = request.dbsession.query(MyModel)
+        one = query.filter(MyModel.name == 'one').first()
+    except DBAPIError:
+        return Response(db_err_msg, content_type='text/plain', status=500)
+    return {'one': one, 'project': 'PyZAPI'}
+
+
+db_err_msg = """\
+Pyramid is having a problem using your SQL database.  The problem
```
Comment (markdia) [inline PyZAPI/pyzapi/views/default.py]: this stuff should be cleaned up and cut out

File: PyZAPI/pyzapi/views/notfound.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from pyramid.view import notfound_view_config
+
+
+@notfound_view_config(renderer='../templates/404.jinja2')
```
Comment (markdia) [inline PyZAPI/pyzapi/views/notfound.py]: are we returning an html 404 here or a proper 404 resource not found rest response?

**Review by markdia — COMMENTED** [review]
Nice work so far on this.  Coming along great.

File: PyZAPI/pyzapi/views/notfound.py (hunk 5 above)
Comment (Nico005) [reply PyZAPI/pyzapi/views/notfound.py]: Hi Mark, This was a default setup (created file) by pyramid, we can change it to a rest 404.

File: PyZAPI/pyzapi/__init__.py (hunk 3 above)
Comment (Nico005) [reply PyZAPI/pyzapi/__init__.py]: default when installing pyramid 1.9

File: PyZAPI/pyzapi/views/default.py (hunk 4 above)
Comment (Nico005) [reply PyZAPI/pyzapi/views/default.py]: this file was the default pyramid view file, it has changed with the inclusion of the API

File: PyZAPI/zappa_settings.json (hunk 1 above)
Comment (Nico005) [reply PyZAPI/zappa_settings.json]: Agreed, sorry this one slipped through will create the example file and push now

File: ansible/roles/base/tasks/main.yml (hunk 2 above)
Comment (Nico005) [reply ansible/roles/base/tasks/main.yml:29]: Yip all good and runs fine without the become: yes

File: PyZAPI/pyzapi/routes.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+def includeme(config):
+    config.add_static_view('static', 'static', cache_max_age=3600)
+    config.add_route('home', '/')
+
+    # relevant routes for the blog post
```
Comment (oakensoul) [inline PyZAPI/pyzapi/routes.py:5]: So, if you’re doing this in a restful manner… It should look more like: I don’t entirely know the framework yet, so you may need to do some tweaking. It’s not the end of the world for this PR, we can change it later.

Single user resoruce = GET /users/{user_id}
Collection of users = GET /users/

Create: POST /users/
Update: PUT /users/{user_id}

Delete: DELETE /users/{user_id}

**Review by oakensoul — COMMENTED** [review]
Looking good, I think it’s good enough for our V1 so that we can start making smaller changes/updates. Let’s see what @ajbosco and @markdia say.

File: PyZAPI/pyzapi/routes.py (hunk 6 above)
Comment (markdia) [reply PyZAPI/pyzapi/routes.py:5]: agreed.  Let's pull this in and then come up with a list of items to iterate forward on.  Be great to get a few more hands on it in this state.

**Review by oakensoul — APPROVED** [review]
Ok, it sounds like everyone is thumbs up with moving on to the next stage.

**Pull request merged**